pinot-tools/src/main/java/com/linkedin/pinot/tools/segment/converter/SegmentMergeCommand.java

```diff
@@ -112,6 +112,23 @@ public boolean execute() throws Exception {
       workingDir = new File(_workingDirectory);
     }
 
+    // Check if the input path is directory or comma separated files
+    String[] paths = _inputSegmentPaths.split(INPUT_SEGMENT_SEPARATOR);
+    List<String> inputPaths = new ArrayList<>();
+    if (paths.length == 1) {
+      File inputDirectory = new File(paths[0].trim());
+      if (inputDirectory.isDirectory()) {
+        for (File path : inputDirectory.listFiles()) {
+          inputPaths.add(path.getAbsolutePath());
+        }
+      }
+    } else {
+      for (String path : paths) {
+        inputPaths.add(path.trim());
```

Comment:
good to check that the path is not a directory here and throw appropriate error. Or, support specifying comma-separated directory and/or paths

Reply:
I changed this to support comma separated files/directories + nested directories. Also, I added to throw the exception if the given input path is not valid.